CMakeLists.txt

```diff
@@ -129,13 +149,20 @@ option(CELERITY_USE_MIMALLOC "Use the mimalloc memory allocator" ${CELERITY_USE_
 include(FetchContent)
 
 macro(fetch_content_from_submodule DEPNAME RELPATH)
-  if(NOT EXISTS "${PROJECT_SOURCE_DIR}/${RELPATH}/CMakeLists.txt")
-    message(FATAL_ERROR "The git submodule ${RELPATH} is missing.\nTry running `git submodule update --init`.")
+  if(${DEPNAME}_FOUND)
```

Comment:
What is this about?

Reply:
SimSYCL transitively pulls in libenvpp to parse its own environment, so this skips Celerity's `fetch_content` if the dependency is already satisfied (by SimSYCL or by a CMake project using Celerity as its dependency for example).

Hoping that the previously found dependency is a match *could* lead to version conflicts which I hope will be debuggable for the user from the `Using system $dep from $path` messages.

Reply:
Ah, I had a similar issue with `fmt` recently - I like it!